### ISSUE TITLE
Opencv4nodejs library isn't connected.

### ISSUE THREAD

**Issue by Maksbjj** [udarrr/opencv4nodejs-prebuilt-install#12]
Hey 
I'm getting the following error when trying to run the code
What could be the issue? 

opencv4nodejs library isn't connected. platform should be x64 - current x64, node12-21 - current v16.19.1, only mac,win,linux - current win32

**Comment by udarrr** [comment]
Hi what's you os ?

**Comment by eduardoabpereira** [comment]
Same here, using windows 11 pro.

opencv4nodejs library isn't connected. platform should be x64 - current x64, node12-21 - current v20.9.0, only mac,win,linux - current win32

**Comment by eduardoabpereira** [comment]
@udarrr do you have any suggestion?

**Comment by studentutu** [comment]
same here - windows 10, electron ~25.0.0, node 18, still 

`opencv4nodejs library isn't connected. platform should be x64 - current x64, node12-21 - current v18.15.0, only mac,win,linux - current win32`

**Comment by studentutu** [comment]
@udarrr  maybe there are some options that we can set to manually trigger initialization for openCV? 

Seems like we can just use options to set it up in \node_modules\opencv4nodejs-prebuilt-install\lib\opencv4nodejs.js
```
(function (exports, require, module, __filename, __dirname, process, global, Buffer) { return function (exports, require, module, __filename, __dirname) { const promisify = require("./promisify");
const extendWithJsSources = require("./src");

const isElectronWebpack =
  // assume module required by webpack if no system path inv envs
  !process.env.path &&
  // detect if electron https://github.com/electron/electron/issues/2288
  global.window &&
  global.window.process &&
  global.window.process.type &&
  global.navigator &&
  (global.navigator.userAgent || "").toLowerCase().indexOf(" electron/") > -1;

let cv;

try {
  cv = isElectronWebpack
    ? require("../build/Release/opencv4nodejs.node")
    : require("./cv");

  // promisify async methods
  cv = promisify(cv);
  cv = extendWithJsSources(cv);
} catch (err) {
  console.log(`opencv4nodejs library isn't connected. platform should be x64 - current ${process.arch}, node12-21 - current ${process.version}, only mac,win,linux - current ${process.platform}`);
}

module.exports = cv;

}.call(this, exports, require, module, __filename, __dirname); });
```

**Comment by studentutu** [comment]
Found out - I receive 
```
Loading non-context-aware native module in renderer
```

**Comment by studentutu** [comment]
Also, when running OpenCV only on main:
```
opencv4nodejs-prebuilt-install\build\Release\opencv4nodejs.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 115. This version of Node.js requires
NODE_MODULE_VERSION 116. Please try re-compiling or re-installing
```

**Comment by studentutu** [comment]
Finally fixed issue with missing openCV - need to manually rebuild with electron-rebuild for a specified electron version and node
add under package.json under

```
{
"scripts":
"opencv-rebuild": "npm rebuild --runtime=electron --target=25.0.1 --disturl=https://electronjs.org/headers --abi=116 --napi-modules opencv4nodejs-prebuilt-install",
    ...
}
``` 

and run it once with `yarn/npm run opencv-rebuild`

Now I still have an issue with buffers inside the main process, as somehow 
`const data = await matRGBA.getDataAsync();`
will crash app with 
`Fatal error in V8: v8_ArrayBuffer_NewBackingStore When the V8 Sandbox is enabled, ArrayBuffer backing stores must be allocated inside the sandbox address space. Please use an appropriate ArrayBuffer::Allocator to allocate these buffers, or disable the sandbox.`

**Comment by studentutu** [comment]
At the end of the day - I will switch to @u4/opencv4nodejs, as it did worked with pnpm as is.

**Comment by udarrr** [comment]
Yes I've tried to support node 21,22 but not-gyp can't build the lib for the versions, node-gyp team is working currently on that, and when it'll be possible i'll retrigger builds, for the time being the tool support only nodejs up to 20, i've released all builds in 4.1.208

**Comment by GreatHongyu** [comment]
> 最后修复了缺少 openCV 的问题 - 需要使用 electron-rebuild 手动重建指定的 electron 版本并 在 package.json 下添加节点
> 
> ```
> {
> "scripts":
> "opencv-rebuild": "npm rebuild --runtime=electron --target=25.0.1 --disturl=https://electronjs.org/headers --abi=116 --napi-modules opencv4nodejs-prebuilt-install",
>     ...
> }
> ```
> 
> 然后运行一次`yarn/npm run opencv-rebuild`
> 
> 现在我仍然对主进程中的缓冲区存在问题，因为不知何故 `const data = await matRGBA.getDataAsync();` 会导致应用程序崩溃 `Fatal error in V8: v8_ArrayBuffer_NewBackingStore When the V8 Sandbox is enabled, ArrayBuffer backing stores must be allocated inside the sandbox address space. Please use an appropriate ArrayBuffer::Allocator to allocate these buffers, or disable the sandbox.`

bro!!!,you save my ass!

**Comment by GreatHongyu** [comment]
> node-gyp 团队目前正在研究这个问题，当可能时我会重新触发构建，目前该工具仅支持 nodejs 到 20，我已经在 4.1.209 中发布了所有构建，并在 package.json 中指定了约束

thank you for your effort
